### PR TITLE
improvement: Allow extracting member when file contains  toplevel `val`|`def`|`given` etc.

### PR DESCRIPTION
**Pull request by jkciesluk**
fixes https://github.com/scalameta/metals/issues/4768

### CODE REVIEW

File: tests/slow/src/test/scala/tests/feature/Scala3CodeActionLspSuite.scala [hunk 1]
```diff
@@ -495,6 +495,45 @@ class Scala3CodeActionLspSuite
     selectedActionIndex = 1,
   )
 
+  checkExtractedMember(
+    "extract-object-main-annot",
+    """|package a
+       |@main def buildSite() =  ???
```
Comment (tgodzik) [inline tests/slow/src/test/scala/tests/feature/Scala3CodeActionLspSuite.scala:501]: Does it actually matter if main annotation is used? If we have any def we should treat the file as one that might contain some toplevel methods. What do you think @ckipp01 ?

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: tests/slow/src/test/scala/tests/feature/Scala3CodeActionLspSuite.scala (hunk 1 above)
Comment (ckipp01) [reply tests/slow/src/test/scala/tests/feature/Scala3CodeActionLspSuite.scala:501]: Yea exactly, even if you have something incredibly simple with no main method like:

```
package example

given String = "hello"

object Foo
```

You should be able tot move `Foo` out into a `Foo.scala`.

**Review by tgodzik — APPROVED** [review]
LGTM!

**Review by tgodzik — CHANGES_REQUESTED** [review]
Actually, I realised the failure in Scala3CodeActionLspSuite is related. Could take a look at it?

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**